encoding/protobuf/protoc-gen-yarpc-go/internal/lib/lib.go

```diff
@@ -642,6 +661,41 @@ func clientServerStreamingMethods(service *protoplugin.Service) ([]*protoplugin.
 	return methods, nil
 }
 
+// fileDescriptorClosureVarName is used to refer to a variable that contains a closure of all encoded
+// filedescriptors required to interpret a specific proto file. It is used in the yarpc codebase to
+// attach reflection information to services.
+func fileDescriptorClosureVarName(f *protoplugin.File) (string, error) {
+	name := f.GetName()
+	if name == "" {
+		return "", fmt.Errorf("Could not create fileDescriptorClosureVarName: %s has no name", f)
+	}
+	h := sha256.Sum256([]byte(name))
+	return fmt.Sprintf("yarpcFileDescriptorClosure%s", hex.EncodeToString(h[:8])), nil
+}
+
+func encodedFileDescriptor(f *protoplugin.File) (string, error) {
+	fdBytes, err := f.SerializedFileDescriptor()
+	if err != nil {
+		return "", err
+	}
+
+	var buf bytes.Buffer
```

Comment:
Could we add an in-line comment for the internal structure of the bytes you are creating here?